src/makeCalculator.js

```diff
@@ -4,7 +4,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(a) {
+      this.result += a;
+    },
+    subtract(a) {
+      this.result -= a;
+    },
+    multiply(a) {
+      this.result *= a;
+    },
+    divide(a) {
+      this.result /= a;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(operation, number) {
+      operation.call(this, number);
+
+      return this;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
The `makeCalculator` function is correctly named according to its purpose. However, the methods within the `calculator` object should use the method shorthand syntax instead of the `function` keyword.